# Probability

In [4]:
P_coffe = 0.65
p_cancer = .005

# p_cancer_given_coffe = P(coffe | Cancer) * P(cancer) / P(Coffee)

p = ( .85 *.005)/ .065
p


0.06538461538461539

In [9]:
from scipy.stats import binom

n = 10
k = 10
p = 9/10

probability = binom.pmf(k, n, p)
probability

0.3486784401000001

In [10]:
from scipy.stats import binom

n = 10
p = 0.9

for k in range(n + 1):
    probability = binom.pmf(k, n, p)
    print("{0} - {1}".format(k, probability))

0 - 9.999999999999981e-11
1 - 8.999999999999986e-09
2 - 3.644999999999997e-07
3 - 8.747999999999995e-06
4 - 0.00013778099999999982
5 - 0.0014880347999999988
6 - 0.011160261000000001
7 - 0.05739562799999998
8 - 0.1937102445
9 - 0.387420489
10 - 0.3486784401000001


In [11]:
from scipy.stats import beta

a = 8
b = 2

p = beta.cdf(.90, a, b)

print(p)

0.7748409780000002


#### So according to our calculation, there is a 77.48% chance the underlying probability of success is 90% or less.

In [12]:
from scipy.stats import beta

a = 8
b = 2

p = 1.0 - beta.cdf(.90, a, b)
print(p)

0.22515902199999982


#### This means that out of 8/10 successful engine tests, there is only a 22.5% chance the underlying success rate is 90% or greater.

In [14]:
from scipy.stats import beta

a = 30
b = 6

p = 1.0 - beta.cdf(.90, a, b)

print(p)

0.13163577484183697


In [15]:
from scipy.stats import beta

a = 8
b = 2

p = beta.cdf(.90, a, b) - beta.cdf(.80, a, b)

print(p)

0.33863336199999994


## Exercise 1:

##### There is a 30% chance of rain today, and a 40% chance your umbrella order will arrive on time. You are eager to walk in the rain today and cannot do so without either! What is the probability it will rain AND your umbrella will arrive?

- P(Rain) = 0.30%
- P(umbrella) = 0.40%
- P(Rain AND Umbrella) = Joint Probability  P(rain) * P(umbrella)
- Joint Probabilities = P(A AND B) = P(A) * P(B)

In [16]:
rain = 0.30
umbrella = 0.40

umbrella * rain

0.12

#### This is known as the sum rule of probability and ensures every joint event is counted only once:
- P(A OR B) = P(A) + P(B) - P(A AND B)

## Exercise 2
#### There is a 30% chance of rain today, and a 40% chance your umbrella order will arrive on time. You will be able to run errands only if it does not rain or your umbrella arrives. What is the probability it will not rain OR your umbrella arrives?

- P(Not Rain) = 0.70%
- P(Umbrella) = 0.40%
- P(Umbrella OR Not Rain) = 0.70 + 0.40 - (0.70 * 0.40)

In [17]:
(0.70 + 0.40) - (0.70 * 0.40)

0.8200000000000001

## Exercise 3
#### There is a 30% chance of rain today, and a 40% chance your umbrella order will arrive on time. However, you found out if it rains there is only a 20% chance your umbrella will arrive on time. What is the probability it will rain AND your umbrella will arrive on time?

- P(rain) = 0.30
- p(umbrella) = 0.40
- P(umbrella | rain) = 0.20
- P(Rain AND Umbrella) =  P(umbrella | Rain) * P(Rain)

In [19]:
(0.20 * 0.30 )

0.06

## Exercise 4

#### You have 137 passengers booked on a flight from Las Vegas to Dallas. However, it is Las Vegas on a Sunday morning and you estimate each passenger is 40% likely to not show up. You are trying to figure out how many seats to overbook so the plane does not fly empty. How likely is it at least 50 passengers will not show up?

In [20]:
from scipy.stats import binom

n = 137  # Total number of passengers
p = 0.40  # Probability of a passenger not showing up

# We want the probability of having at least 50 passengers not show up
k = 50

# Calculating P(X >= 50)
# Since CDF gives P(X <= k), we use 1 - P(X <= k-1) to get P(X >= k)
probability_at_least_50 = 1 - binom.cdf(k-1, n, p)

print(f"The probability that at least 50 passengers will not show up is: {probability_at_least_50:.4f}")


The probability that at least 50 passengers will not show up is: 0.8221


## Exercise 5
#### You flipped a coin 10 times and got heads 8 times and tails 2 times. Do you think this coin has any good probability of being fair? Why or why not?

In [22]:
from scipy.stats import beta

a = 8
b = 2

1 - beta.cdf(.50, a, b)

0.98046875